lib/apiservers/engine/backends/backends.go

```diff
@@ -197,18 +201,56 @@ func InsecureRegistries() []string {
 }
 
 // syncContainerCache runs once at startup to populate the container cache
-func syncContainerCache(client *client.PortLayer) error {
+func syncContainerCache() error {
 	log.Debugf("Sync up container cache from portlyaer")
 
+	backend := NewContainerBackend()
+	client := backend.containerProxy.Client()
 	all := true
 	containme, err := client.Containers.GetContainerList(containers.NewGetContainerListParamsWithContext(ctx).WithAll(&all))
 	if err != nil {
 		return errors.Errorf("Failed to retrieve container list from portlayer: %s", err)
 	}
+
 	cc := cache.ContainerCache()
+	var errs []string
 	for _, info := range containme.Payload {
 		container := ContainerInfoToVicContainer(*info)
 		cc.AddContainer(container)
+		if err = setPortMapping(info, backend, container); err != nil {
+			errs = append(errs, err.Error())
+		}
+	}
+	if len(errs) > 0 {
+		return errors.Errorf("Failed to set port mapping: %s", strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
+func setPortMapping(info *models.ContainerInfo, backend *Container, container *container.VicContainer) error {
+	if info.ContainerConfig.State == nil {
+		log.Infof("container state is nil")
+		return nil
+	}
+	if *info.ContainerConfig.State != "Running" || len(container.HostConfig.PortBindings) == 0 {
+		log.Infof("Container info state: %s", *info.ContainerConfig.State)
+		log.Infof("container portbinding: %+v", container.HostConfig.PortBindings)
+		return nil
+	}
+
+	log.Debugf("Set port mapping for container %q, portmapping %+v", container.Name, container.HostConfig.PortBindings)
+	client := backend.containerProxy.Client()
+	endpointsOK, err := client.Scopes.GetContainerEndpoints(
```

Comment:
this call is to get back endpoint information, including the containerIP, which is not persisted in portmapping.
Need this information if container is running